### PR TITLE
Add an option allowing to treat lower bound zero commit as any commit

### LINKED ISSUE #154: Add an option allowing to treat lower bound zero commit as any commit
Hello,

`from` commit is excluded from the list of commits unless the former is the commit zero:
https://github.com/tomasbjerre/git-changelog-lib/blob/d5a14eca9ffed77e258c3c2e4a12c93c74a2f73a/src/main/java/se/bjurr/gitchangelog/internal/git/GitRepo.java#L253-L255

Because of that, you can't only retrieve the 2nd and 3rd commit of a repository. You can either:
* retrieve 1st, 2nd and 3rd
* retrieve the 3rd

I'd like an option to disable commit zero special treatment.

### PR DESCRIPTION
**Pull request by reda-alaoui**
Fix #154

### CODE REVIEW

**Comment by tomasbjerre** [comment]
I think I would rather have a way of specifying if "from" is inclusive or not. And if "to" is inclusive or not. Can be two different booleans.

These methods:
https://github.com/tomasbjerre/git-changelog-lib/blob/e8e15c750a5f8c1be34019a1453468374c0ec58f/src/main/java/se/bjurr/gitchangelog/api/GitChangelogApi.java#L306
Could take a boolean that specifies it it is inclusive or exclusive.

Another, perhaps better, alternative might be if there are from/to -methods where a user dont need to specify if it is a ref or commit. It would just be "from(String revision, boolean inclusive)" and another method for "to".

**Comment by reda-alaoui** [comment]
@tomasbjerre , should I try to keep backward compatibility?

**Comment by reda-alaoui** [comment]
> Could take a boolean that specifies it it is inclusive or exclusive.

If I want to keep backward compatibility, this could be a nullable Boolean. If null, it'll keep the current behaviour. Otherwise include or exclude the commit.

**Comment by tomasbjerre** [comment]
I merged the `fromRevision` and `toRevision` part of this to master. And also made an extra commit to fix some test cases. Just a heads up to rebase and get a cleaner diff here.

File: src/test/resources/settings/git-changelog-test-settings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
  "fromRepo": ".",
- "fromRevision": "0000000000000000000000000000000000000000",
```
Comment (tomasbjerre) [inline src/test/resources/settings/git-changelog-test-settings.json:3]: `fromRevision` and `toRevision` should be kept as is here right?

Comment (reda-alaoui) [reply src/test/resources/settings/git-changelog-test-settings.json:3]: Yes. I just fixed, I guess.

**Comment by tomasbjerre** [comment]
> @tomasbjerre , I guess I should add some tests. Could you give me some pointers?

There is a tag called `test` and some test cases use that tag to test parsing:
https://github.com/tomasbjerre/git-changelog-lib/commits/test

Perhaps use that to parse the commits and make assertions:

```java
    Changelog actual = gitChangelogApiBuilder() //
	    .withFromRevision(ZERO_COMMIT) //
	    .withToRevision("test") //
	    .getChangelog();

    assertThat(actual.getCommits()).hasSize(123);
    assertThat(actual.getCommits().stream().map(Commit::getHash).collect(Collectors.joining(","))).isEqualTo("asdasdasdas");
```

Perhaps you write the test cases in a new Test class, perhaps `BoundaryTest`.

File: src/main/java/se/bjurr/gitchangelog/internal/git/GitRepo.java [hunk 2]
```diff
@@ -238,22 +237,8 @@ private Map<String, RevTag> getAnnotatedTagPerTagName(
...
-    while (itr.hasNext()) {
-      list.add(itr.next());
-    }
-    this.revWalk.parseHeaders(from);
-    if (from.getParentCount() == 0) {
-      list.add(from);
-    }
-    return list;
+  private List<RevCommit> getCommitList(final RevCommitBoundary from, final RevCommitBoundary to) throws Exception {
+    return RevCommitBoundary.listCommits(git, revWalk, from, to, pathFilter);
```
[the remark this answers is not in the document]
Comment (reda-alaoui) [reply src/main/java/se/bjurr/gitchangelog/internal/git/GitRepo.java]: I reduced the diff.

**Comment by reda-alaoui** [comment]
Tests added

**Comment by reda-alaoui** [comment]
@tomasbjerre I improved the tests. In the process, I discovered a little bug fixed by https://github.com/tomasbjerre/git-changelog-lib/pull/156/commits/a27dc05376eb02733d524f8ed52a9513fc957a4e#diff-3129b9a512e39549cbae022695e97402365f3585d43779d5da398cd21aa870ffR484 .

Waiting for you review :pray:

**Comment by tomasbjerre** [comment]
thanks

**Comment by reda-alaoui** [comment]
@tomasbjerre could you please draft a new release?

**Comment by tomasbjerre** [comment]
I made some adjustments and released 1.171.0.

**Pull request merged**